task_processing/plugins/mesos/retrying_executor.py

```diff
@@ -108,3 +118,9 @@ def stop(self):
 
     def get_event_queue(self):
         return self.dest_queue
+
+    def grow(self, task_id):
+        return task_id.append(self.retries - self.task_retries[task_id])
+
+    def shrink(self, task_id):
+        return task_id[0:-1]
```

Comment:
I am not sure about assuming a special format of task_id (e.g. X:Y:Z) is a robust way to deal with delayed/duplicated updates. Users now have a choice to pass in task_id without control. ':' may be included in a user generated task_id. The bullet proof way is to keep track of retried task_id, more work, but safer.